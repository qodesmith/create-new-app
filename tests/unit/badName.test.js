const badName = require('../../modules/badName')


describe('badName', () => {
  const originalConsole = console.log
  const naughtyName = 'nope! no way!'
  const validation = {
    errors: ['no bueno'],
    warnings: ['naughty naughty']
  }
  console.log = jest.fn()

  beforeEach(() => badName(naughtyName, validation))
  afterEach(() => console.log.mockReset())
  afterAll(() => console.log = originalConsole)

  it('should log a message with the failed app name, errors, & warnings', () => {
    expect(console.log).toHaveBeenCalledTimes(3)
    expect(console.log.mock.calls).toMatchSnapshot()
  })
})
