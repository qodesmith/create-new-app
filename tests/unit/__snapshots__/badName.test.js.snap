// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badName should log a message with the failed app name, errors, & warnings 1`] = `
Array [
  Array [
    "Could not create a project called [31m\\"nope! no way!\\"[39m because of npm naming restrictions:",
  ],
  Array [
    "[31m  *  no bueno[39m",
  ],
  Array [
    "[33m  *  naughty naughty[39m",
  ],
]
`;
